streams/test-utils/src/main/java/org/apache/kafka/streams/processor/MockProcessorContext.java

```diff
@@ -0,0 +1,465 @@
+     */
+    public static class CapturedPunctuator {
+        private final long intervalMs;
+        private final PunctuationType type;
+        private final Punctuator punctuator;
+
+        // mutable:
+        private boolean cancelled = false;
+
+        private CapturedPunctuator(final long intervalMs, final PunctuationType type, final Punctuator punctuator) {
+            this.intervalMs = intervalMs;
+            this.type = type;
+            this.punctuator = punctuator;
+        }
+
+        public long getIntervalMs() {
+            return intervalMs;
+        }
+
+        public PunctuationType getType() {
+            return type;
+        }
+
+        public Punctuator getPunctuator() {
+            return punctuator;
+        }
+
+        public void cancel() {
+            this.cancelled = true;
+        }
+
+        public boolean cancelled() {
+            return cancelled;
+        }
+    }
+
+    private final List<CapturedPunctuator> punctuators = new LinkedList<>();
+
+    public static class CapturedForward {
+        /*Nullable*/ private final String childName;
+        private final long timestamp;
+        private final KeyValue kv;
+
+        private CapturedForward(final To to, final KeyValue kv) {
+            if (kv == null) throw new IllegalArgumentException();
+
+            this.childName = to.childName;
+            this.timestamp = to.timestamp;
+            this.kv = kv;
+        }
+
+        /**
+         * The child this data was forwarded to.
+         * @return The child name, or {@code null} if it was broadcasted.
+         */
+        public String childName() {
+            return childName;
+        }
+
+        /**
+         * The timestamp attached to the forwarded record.
+         * @return A timestamp, or {@code -1} if none was forwarded.
+         */
+        public long timestamp() {
+            return timestamp;
+        }
+
+        /**
+         * The data forwarded.
+         * @return A key/value pair. Not null.
+         */
+        public KeyValue kv() {
+            return kv;
+        }
+    }
+
+    private List<CapturedForward> capturedForwards = new LinkedList<>();
+
+    private boolean committed = false;
+
+    // contructors ================================================
+
+    /**
+     * Create a {@link MockProcessorContext} with dummy {@code config} and {@code taskId} and {@code null} {@code stateDir}.
+     * Most unit tests using this mock won't need to know the taskId,
+     * and most unit tests should be able to get by with the
+     * {@link InMemoryKeyValueStore}, so the stateDir won't matter.
+     */
+    public MockProcessorContext() {
+        //noinspection DoubleBraceInitialization
+        this(
+            new Properties() {
+                {
+                    put(StreamsConfig.APPLICATION_ID_CONFIG, "");
+                    put(StreamsConfig.BOOTSTRAP_SERVERS_CONFIG, "");
```

Comment:
Ok, I'l take out the url.

Reply:
yes, both configs are required.

Reply:
Where does it fail if we don't specify them? In KS they are required, because we pass the configs into a `StreamsConfig` -- but we don't use `StreamsConfig` (if I did not miss this) thus I think we can simplify the code here and just pass in empty `Properties`

Reply:
```
org.apache.kafka.common.config.ConfigException: Missing required configuration "application.id" which has no default value.

	at org.apache.kafka.common.config.ConfigDef.parseValue(ConfigDef.java:472)
	at org.apache.kafka.common.config.ConfigDef.parse(ConfigDef.java:462)
	at org.apache.kafka.common.config.AbstractConfig.<init>(AbstractConfig.java:62)
	at org.apache.kafka.common.config.AbstractConfig.<init>(AbstractConfig.java:75)
	at org.apache.kafka.streams.StreamsConfig.<init>(StreamsConfig.java:703)
	at org.apache.kafka.streams.processor.MockProcessorContext.<init>(MockProcessorContext.java:190)
	at org.apache.kafka.streams.MockProcessorContextTest.fullConstructorShouldSetAllExpectedAttributes(MockProcessorContextTest.java:415)
```
and
```
org.apache.kafka.common.config.ConfigException: Missing required configuration "bootstrap.servers" which has no default value.

	at org.apache.kafka.common.config.ConfigDef.parseValue(ConfigDef.java:472)
	at org.apache.kafka.common.config.ConfigDef.parse(ConfigDef.java:462)
	at org.apache.kafka.common.config.AbstractConfig.<init>(AbstractConfig.java:62)
	at org.apache.kafka.common.config.AbstractConfig.<init>(AbstractConfig.java:75)
	at org.apache.kafka.streams.StreamsConfig.<init>(StreamsConfig.java:703)
	at org.apache.kafka.streams.processor.MockProcessorContext.<init>(MockProcessorContext.java:190)
	at org.apache.kafka.streams.MockProcessorContextTest.fullConstructorShouldSetAllExpectedAttributes(MockProcessorContextTest.java:415)
```